axelor-account/src/main/resources/views/Period.xml

```diff
@@ -25,4 +42,16 @@
     <call class="com.axelor.apps.account.web.PeriodController" method="continueClose"/>
   </action-method>
 
+  <action-method name="action-method-account-period-control-dates">
+    <call class="com.axelor.apps.account.web.PeriodController" method="controlDates"/>
+  </action-method>
+
+  <action-attrs name="action-attrs-account-readonly-if-period-in-move">
+    <attribute name="readonly" for="fromDate"
+      expr="eval: (__repo__(Move).all().filter('self.period = ?', __this__).count() > 0) &amp;&amp; statusSelect == 1 &amp;&amp; year?.statusSelect == 1"/>
+    <attribute name="readonly" for="toDate"
+      expr="eval: (__repo__(Move).all().filter('self.period = ?', __this__).count() > 0) &amp;&amp; statusSelect == 1 &amp;&amp; year?.statusSelect == 1"/>
+    <attribute name="readonly" for="year"
+      expr="eval: (__repo__(Move).all().filter('self.period = ?', __this__).count() > 0) &amp;&amp; statusSelect == 1 &amp;&amp; year?.statusSelect == 1"/>
```

Comment:
I think this way, the query will be run three times. Should be moved to a java method to only run the query once and use the result for each field.